### PR TITLE
Add r-duckdb recipe via conda-r-skeleton-helper

### PR DESCRIPTION
**Pull request by izahn**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by izahn** [comment]
I would really appreciate it if you have the time to look at the osx build failure here @conda-forge/help-r` @conda-forge/help-c-cpp

It says 
```
duckdb/src/common/file_system.cpp:349:52: error: use of undeclared identifier '_SC_PHYS_PAGES'
        idx_t max_memory = MinValue<idx_t>((idx_t)sysconf(_SC_PHYS_PAGES) * (idx_t)sysconf(_SC_PAGESIZE), UINTPTR_MAX);
                                                          ^
1 error generated.
```
but I'm not sure what to do about that. Any suggestions appreciated!

File: recipes/r-duckdb/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+export DISABLE_AUTOBREW=1
+
+# R refuses to build packages that mark themselves as Priority: Recommended
+mv DESCRIPTION DESCRIPTION.old
+grep -va '^Priority: ' DESCRIPTION.old > DESCRIPTION
+
+export MACOSX_VERSION_MIN="10.13"
```
Comment (xhochy) [inline recipes/r-duckdb/build.sh:9]: You probably need https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=macosx_sdk_version#requiring-newer-macos-sdks

Comment (izahn) [reply recipes/r-duckdb/build.sh:9]: Thanks, I don't know how I missed that in the documentation :facepalm:

**Comment by izahn** [comment]
As per the documentation at https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=macosx_sdk_version#requiring-newer-macos-sdks I'm pinging @conda-forge/core because I think I need to set `MACOSX_SDK_VERSION` to 10.12 in order to get this recipe to build.

**Review by xhochy — APPROVED** [review]
LGTM but @conda-forge/help-r should push the final merge button.

File: recipes/r-duckdb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  license: MPL-2.0
+  summary: The DuckDB project is an embedded analytical data management system with support for
+    the Structured Query Language (SQL). This package includes all of DuckDB and a R
+    Database Interface (DBI) connector.
+  license_family: MOZILLA
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - izahn
```
Comment (cbrueffer) [inline recipes/r-duckdb/meta.yaml:62]:
```suggestion
    - conda-forge/r
    - izahn
```

File: recipes/r-duckdb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  sha256: 13ba53b22dbcda041d57f3d15ec96eae5a98e8529a031e3b584629515dc51918
+
+build:
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+# Suggests: callr, DBItest, dbplyr, nycflights13, testthat, withr
```
Comment (cbrueffer) [inline recipes/r-duckdb/meta.yaml:23]:
```suggestion
```

**Review by cbrueffer — COMMENTED** [review]
Some small things; please always include `conda-forge/r` as maintainer for R packages.

**Comment by izahn** [comment]
Thank you @cbrueffer that's twice today you fixed things up for me. I do appreciate it!

**Comment by cbrueffer** [comment]
No problem, it was all pretty minor stuff.

**Pull request merged**